src/App.js

```diff
@@ -74,6 +86,20 @@ export default class App extends React.Component {
 		this.setState({ savedCollections: newSavedCollections });
 		this.setInStorage('savedCollections', newSavedCollections);
 	};
+	showCarousel = (photo) => {
+		let index = this.state.savedPhotos.findIndex((i) => i.id === photo.id);
+		this.setState({
+			carousel: [
+				...this.state.savedPhotos.slice(index),
+				...this.state.savedPhotos.slice(0, index),
+			],
+			display: 'block',
+		});
+	};
+
+	closeCarousel = () => {
+		this.setState({ display: 'none' });
+	};
```

Comment:
The carousel will begin from the savedPhotos index that we clicked on

Reply:
maybe set state true false and change the display in your styled component